src/views/product-downloads-view/types.ts

```diff
@@ -1,40 +1,77 @@
-import { ReleasesAPIResponse } from 'lib/fetch-release-data'
+import { ReleasesAPIResponse, GeneratedProps } from 'lib/fetch-release-data'
 import { VersionContextSwitcherProps } from 'components/version-context-switcher'
 import { SidecarMarketingCardProps } from './components/sidecar-marketing-card'
+import { CollectionCardPropsWithId } from 'components/collection-card'
+import { TutorialCardPropsWithId } from 'components/tutorial-card'
 
-export interface FeaturedTutorial {
-  description: string
-  href: string
-  title: string
-}
-
-export interface GroupedPackageManagers {
-  [os: string]: PackageManager[]
-}
+/**
+ * Raw page content, as structured in .json files
+ */
+export type FeaturedTutorialContent =
+  | {
+      collectionSlug: string
+      tutorialSlug: never
+    }
+  | {
+      tutorialSlug: string
+      collectionSlug: never
+    }
 
 export interface PackageManager {
   label: string
   commands: string[]
   os: string
 }
 
-export interface ProductDownloadsViewContentProps {
-  pageContent: ProductDownloadsViewProps['pageContent']
-  releases: ProductDownloadsViewProps['releases']
-  versionSwitcherOptions: VersionContextSwitcherProps['options']
+export interface RawProductDownloadsViewContent {
+  doesNotHavePackageManagers?: boolean
+  featuredTutorials?: FeaturedTutorialContent[]
+  packageManagerOverrides?: PackageManager[]
+  sidecarMarketingCard: SidecarMarketingCardProps
 }
 
+/**
+ * View prop types
+ */
+
+export type FeaturedTutorialCard =
+  | ({ type: 'collection' } & CollectionCardPropsWithId)
+  | ({ type: 'tutorial' } & TutorialCardPropsWithId)
+
 export interface ProductDownloadsViewProps {
   latestVersion: string
   pageContent: {
     doesNotHavePackageManagers?: boolean
-    featuredTutorials?: FeaturedTutorial[]
+    featuredTutorialCards?: FeaturedTutorialCard[]
     packageManagerOverrides?: PackageManager[]
     sidecarMarketingCard: SidecarMarketingCardProps
   }
```

Comment:
💭 I think we probably want to update this to extend `RawProductDownloadsViewContent`, to make the relationship between `RawProductDownloadsViewContent` and `ProductDownloadsViewProps['pageContent']` clearer.

```tsx
// this would be new
interface ProductDownloadsViewPageContentProp
  extends RawProductDownloadsViewContent {
  featuredTutorialCards?: FeaturedTutorialCard[]
}

export interface ProductDownloadsViewProps {
  latestVersion: string
  pageContent: ProductDownloadsViewPageContentProp
  releases: ReleasesAPIResponse
}
```

Reply:
Personally I find this adds another layer of complexity to some already complex types - I think to be fully accurate we'd need to do:

```ts
interface ProductDownloadsViewPageContentProp
  extends Omit<RawProductDownloadsViewContent, 'featuredTutorials'> {
  featuredTutorialCards?: FeaturedTutorialCard[]
}

export interface ProductDownloadsViewProps {
  latestVersion: string
  pageContent: ProductDownloadsViewPageContentProp
  releases: ReleasesAPIResponse
}
```

Strong opinion weakly held, but I think it's easier to figure out the shape of props here without these extra layers.

Reply:
I see what you're saying! `ProductDownloadsViewPageContentProp` would also reduce some duplication, but fine with deferring that for now.

---

Just for the sake of discussion (no action needed), I think these two have the same output as the code currently stands so I think it becomes a matter of preference?

```ts
interface ProductDownloadsViewPageContentProp
  extends RawProductDownloadsViewContent {
  featuredTutorialCards?: FeaturedTutorialCard[]
}

interface ProductDownloadsViewPageContentProp
  extends Omit<RawProductDownloadsViewContent, 'featuredTutorials'> {
  featuredTutorialCards?: FeaturedTutorialCard[]
}
```

Reply:
☝️ Great question! Definitely optional 💭 thought here, but here's my take:

---

In the first example above, without `Omit`, I think it's possible that `ProductDownloadsViewPageContentProp` would have an optional property `featuredTutorialContent` (fka `featuredTutorials`)? But I think we might want to make sure that property isn't present in the type, for clarity and to avoid confusion.

To be more concrete, without `Omit`, it's possible to include `featuredLearnContent` slugs, even though they won't be used on the front end:

<img width="1041" alt="without-omit" src="https://user-images.githubusercontent.com/4624598/171320195-148951ef-5931-4308-bcc5-1eef022b3aa3.png">

With `Omit` this gets flagged (it won't be used on the front-end, only `featuredLearnCards` will be, so it feels worth flagging to me): 

<img width="1122" alt="with-omit" src="https://user-images.githubusercontent.com/4624598/171320201-ba8eb292-a218-4d21-bce7-8b5b08c56da9.png">

---

More broadly, I think what's meant by `output` is maybe a key there. Personally, one "output" of type files for me is of course what's actually happening when types are being checked (and `Omit` feels important to me there). 

Another though is being able to look at the type file itself, and at type annotations, and hover or go-to-definitions etc and get meaningful information as quickly as I can (so that I don't get thrown out of whatever context I'm in). I find the extra layer means I have to "go to definition" an extra time to get meaningful information:

![CleanShot 2022-05-31 at 23 18 45@2x](https://user-images.githubusercontent.com/4624598/171321479-e2fc03d2-0411-4336-816d-759a69f83657.png)

Where being slightly repetitive and explicit surfaces things more quickly:

![CleanShot 2022-05-31 at 23 21 19@2x](https://user-images.githubusercontent.com/4624598/171321459-1728c67b-23f3-422f-af2c-fae9cf1b1e3a.png)

Definitely agreed it feels like a matter of preference though... I think in this case, unlike with `Omit`, the type checking will shake out the exact same. Personally I find a bit of repetition is worth it for types that feel a bit more easy-to-dig-through. I think it's about the same amount of type-safe either way, so personally I would rather leave the types explicit and keep the abstraction and engineering of types to a minimum.